Microsoft.DotNet.ILCompiler build fail

### Description

Last console output is 

```
C:\Users\Administrator\.nuget\packages\htmlagilitypack\1.11.42\lib\netstandard2.0\HtmlAgilityPack.dll : warning IL3053:
 Assembly 'HtmlAgilityPack' produced AOT analysis warnings. [G:\gitsouce\Anf\Platforms\Anf.Web\Anf.Web.csproj]
EXEC : error : Object reference not set to an instance of an object. [G:\gitsouce\Anf\Platforms\Anf.Web\Anf.Web.csproj]
  System.NullReferenceException: Object reference not set to an instance of an object.
     at ILCompiler.DependencyAnalysis.ConstrainedMethodUseLookupResult.GetTarget(NodeFactory factory, GenericLookupResu
  ltContext dictionary)
     at ILCompiler.DependencyAnalysis.ReadyToRunGenericHelperNode.InstantiateDependencies(NodeFactory factory, Instanti
  ation typeInstantiation, Instantiation methodInstantiation)
     at ILCompiler.DependencyAnalysis.ShadowConcreteMethodNode.GetStaticDependencies(NodeFactory factory)
     at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.GetStaticDependenciesImpl(DependencyNodeCore`1 node
  )
     at ILCompiler.DependencyAnalysisFramework.DependencyAnalyzer`2.ComputeMarkedNodes()
     at ILCompiler.ILScanner.ILCompiler.IILScanner.Scan()
     at ILCompiler.Program.Run(String[] args)
     at ILCompiler.Program.Main(String[] args)
C:\Users\Administrator\.nuget\packages\microsoft.dotnet.ilcompiler\7.0.0-preview.3.22175.4\build\Microsoft.NETCore.Nati
ve.targets(269,5): error MSB3073: 命令“"C:\Users\Administrator\.nuget\packages\runtime.win-x64.microsoft.dotnet.ilcompile
r\7.0.0-preview.3.22175.4\tools\\ilc" @"obj\Release\net7.0\win-x64\native\Anf.Web.ilc.rsp"”已退出，代码为 1。 [G:\gitsouce\Anf\
Platforms\Anf.Web\Anf.Web.csproj]
```

### Reproduction Steps

Use command

```cmd
dotnet publish -c Release -f net6.0 -r win-x64
```

### Expected behavior

Fail to compile to native

### Actual behavior

Succeed

### Regression?

.NET 6 or .NET 7

Microsoft.DotNet.ILCompiler 7.0.0-preview.3.22175.4

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

_No response_

--- a/src/coreclr/tools/Common/Compiler/TypeExtensions.cs
+++ b/src/coreclr/tools/Common/Compiler/TypeExtensions.cs
@@ -682,6 +682,28 @@ namespace ILCompiler
                 if (c.ContainsSignatureVariables())
                     return null;
 
+                // If there's unimplemented static abstract methods, this is not a suitable instantiation.
+                // We shortcut to look for any static virtuals. It matches what Roslyn does for error CS8920.
+                // Once TypeSystemConstraintsHelpers is updated to check constraints around static virtuals,
+                // we could dispatch there instead.
+                if (c.IsInterface)
+                {
+                    if (HasStaticVirtualMethods(c))
+                        return null;
+
+                    foreach (DefType intface in c.RuntimeInterfaces)
+                        if (HasStaticVirtualMethods(intface))
+                            return null;
+
+                    static bool HasStaticVirtualMethods(TypeDesc type)
+                    {
+                        foreach (MethodDesc method in type.GetVirtualMethods())
+                            if (method.Signature.IsStatic)
+                                return true;
+                        return false;
+                    }
+                }
+
                 constrainedType = c;
             }
 

